lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1594,7 +1608,7 @@ def _set_scrape_job_spec(self, event):
                 # that is written to the filesystem.
                 relation.data[self._charm.app]["alert_rules"] = json.dumps(alert_rules_as_dict)
 
-    def _set_unit_ip(self, _):
+    def _set_unit_ip(self, _=None):
```

Comment:
Did this have a functional effect, or was it just for style?  Haven't seen it before so just curious what prompted it

Reply:
Yes. Unit tests for Prometheus were broken. There were many other changes in the Prometheus lib regarding when to update scrape jobs. In unit tests, I had to move relation creation before charm init. Does not make much sense to me. I pinged the observability team, but they did not respond on this.